src/main/java/frc/robot/subsystems/Drivetrain.java

```diff
@@ -0,0 +1,205 @@
+
+    m_leftMotors = new SpeedControllerGroup(m_leftMotor1, m_leftMotor2);
+    m_rightMotors = new SpeedControllerGroup(m_rightMotor1, m_rightMotor2);
+
+    m_Drive = new DifferentialDrive(m_leftMotors, m_rightMotors);
+
+     // adds options to the drive type
+     m_chooser.addOption("Tank Drive", "Tank");
+     m_chooser.addOption("Arcade Drive", "Arcade");
+     m_chooser.addOption("Trigger Drive", "Trigger");
+
+    m_leftMotorEncoder1 = m_leftMotor1.getEncoder();
+    m_leftMotorEncoder2 = m_leftMotor2.getEncoder();
+    m_rightMotorEncoder1 = m_rightMotor1.getEncoder();
+    m_rightMotorEncoder2 = m_rightMotor2.getEncoder();
+
+    motorInit(m_rightMotor1, DriveConstants.kRightEncoderInverted);
+    motorInit(m_rightMotor2, DriveConstants.kRightEncoderInverted);
+    motorInit(m_leftMotor1, DriveConstants.kLeftEncoderInverted);
+    motorInit(m_leftMotor2, DriveConstants.kLeftEncoderInverted);
+
+
+    addChild("Left Motors", m_leftMotors);
+    addChild("Right Motors", m_rightMotors);
+    addChild("Drivetrain", m_Drive);
+
+  }
+
+private void encoderInit(CANEncoder encoder) {
+    // Converts the input into desired value for distance and velocity
+    encoder.setPositionConversionFactor(DriveConstants.kEncoderDistancePerPulse); // ADD kEncoderDistancePerPulse TO CONSTANTS
+    encoder.setVelocityConversionFactor(DriveConstants.kEncoderSpeedPerPulse); // ADD kEncoderSpeedPerPulse TO CONSTANTS
+    encoderReset(encoder);
+  }
+
+private void encoderReset(CANEncoder encoder) {
+    encoder.setPosition(0);
+  }
+
+public void resetLeftEncoders() {
+    // Reset all left-side drive encoders
+    encoderReset(m_leftMotorEncoder1);
+    encoderReset(m_leftMotorEncoder2);
+  }
+
+public void resetRightEncoders() {
+    // Reset all right-side drive encoders
+    encoderReset(m_rightMotorEncoder1);
+    encoderReset(m_rightMotorEncoder2);
+  }
+
+public void resetAllEncoders() {
+    // Reset all drive encoders
+    resetLeftEncoders();
+    resetRightEncoders();
+  }
+
+private void motorInit(CANSparkMax motor, boolean invert) {
+    motor.restoreFactoryDefaults(); // Reset settings in motor in case they are changed
+    motor.setIdleMode(IdleMode.kBrake); // Sets the motors to brake mode from the beginning
+
+    encoderInit(motor.getEncoder()); // Initializes encoder within motor
+  }
+
+// Gets left side distance by averaging left encoders
+private double getLeftEncoderAverage() {
+    return (m_leftMotorEncoder1.getPosition() + m_leftMotorEncoder2.getPosition()) / 2.0;
+  }
+
+// Gets right side distance by averaging right encoders
+private double getRightEncoderAverage() {
+    return (m_rightMotorEncoder1.getPosition() + m_rightMotorEncoder2.getPosition()) / 2.0;
+  }
+
+// Gets average distance by averaging the right and left distance
+public double getDistance() {
+    return (getRightEncoderAverage() + getLeftEncoderAverage()) / 2.0;
+  }
```

Comment:
We should evaluate in the near future if we feel this is the best way to do it. Maybe a Chief Delphi search?